torch/testing/_internal/common_methods_invocations.py

```diff
@@ -837,6 +837,56 @@ def sample_inputs_reduction(op_info, device, dtype, requires_grad, **kwargs):
     return inputs
 
 
+def _generate_masked_reduction_mask(input_shape, device, requires_grad, **kwargs):
+    yield None
+    yield make_tensor(input_shape, device, torch.bool, requires_grad=requires_grad)
+    # TODO: generate a mask that would lead to broadcasting
+
+
+def sample_inputs_masked_reduction(op_info, device, dtype, requires_grad, **kwargs):
+    """Sample inputs for masked reduction operators."""
+    inputs: List[SampleInput] = []
+    for sample_input in sample_inputs_reduction(op_info, device, dtype, requires_grad, **kwargs):
+        for mask in _generate_masked_reduction_mask(sample_input.input.shape, device, requires_grad, **kwargs):
+            inputs.append(SampleInput(sample_input.input,
+                                      args=(mask,) + sample_input.args,
+                                      kwargs=sample_input.kwargs))
+    return inputs
+
+
+def reference_masked_sum(input, mask, dim=None, keepdim=False, dtype=None):
+    """Reference implementation of masked_sum.
+
+    See RFC-0016 for more information.
```

Comment:
For convenience and forward compatibility it could help to add the explicit link ```https://github.com/pytorch/rfcs/pull/27```

Reply:
reference_masked_sum is removed.